src/test_l1_dsDisplay.c

```diff
@@ -641,51 +625,40 @@ void test_l1_dsDisplay_positive_dsGetEDIDBytes(void) {
     unsigned char *edid1 = (unsigned char *)malloc(512);
     unsigned char *edid2 = (unsigned char *)malloc(512);
     int length1 = 0;
-    int length2 = 0;
 
     // Step 01: Initialize the display sub-system
     result = dsDisplayInit();
     UT_LOG("\n In %s Return value: [%d]\n", __FUNCTION__, result);
     UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-    // Loop through all valid ports
-    for (int i = 0; i < sizeof(kSupportedPortTypes) / sizeof(kSupportedPortTypes[0]); i++) {
-        // Step 02: Call dsGetDisplay() for each valid port
-        result = dsGetDisplay(kSupportedPortTypes[i], i, &displayHandle);
+    // Step 02: Call dsGetDisplay() for each valid port
+    uint32_t numPorts = UT_KVP_PROFILE_GET_UINT32("dsDisplay/Number_of_ports");
+    uint32_t videoPorts = UT_KVP_PROFILE_GET_UINT32("dsDisplay/Video_Ports");
+    for (size_t i = 0; i < numPorts; i++) {
+        uint32_t vType = videoPorts;
+        result = dsGetDisplay(vType, 0, &displayHandle);
         if(result != dsERR_NONE)
         {
             UT_FAIL("Failed to get display handle");
-            break;
+            return;
```

Comment:
function should not return here, we are exiting without term and freeing the buffers.
use UT_ASSERT_EQUAL. Same for line 649

Reply:
done